src/documents/firstDiscoveryServer.md

```diff
@@ -0,0 +1,383 @@
+                The path to the resource for creating a preference set.
+                A query parameter can be added to provide the view/ontology that the
+                preferences are to be stored in.
+                <p>
+                    Example:
+                    "/add-preferences?view=%view"
+                </p>
+                <p>
+                    Option Path:
+                    <code>"preferencesConfig.securityServer.paths.preferences"</code>
+                </p>
+            </td>
+        </tr>
+        <tr>
+            <td>
+                <code>GPII_OAUTH2_AUTH_GRANT_TYPE</code>
+            </td>
+            <td>
+                The grant type supported by the OAuth2 server.
+                <p>
+                    Example:
+                    "client_credentials"
+                </p>
+                <p>
+                    Option Path:
+                    <code>"preferencesConfig.authentication.grant_type"</code>
+                </p>
+            </td>
+        </tr>
+        <tr>
+            <td>
+                <code>GPII_OAUTH2_AUTH_SCOPE</code>
+            </td>
+            <td>
+                The level of permissions that are being requested.
+                <p>
+                    Example:
+                    "add_preferences"
+                </p>
+                <p>
+                    Option Path:
+                    <code>"preferencesConfig.authentication.scope"</code>
+                </p>
+            </td>
+        </tr>
+        <tr>
+            <td>
+                <code>GPII_OAUTH2_AUTH_CLIENT_ID</code>
+            </td>
+            <td>
+                The client ID registered with the OAuth2 server.
+                Should be kept confidentially.
+                <p>
+                    Example:
+                    "client_id_first_discovery"
+                </p>
+                <p>
+                    Option Path:
+                    <code>"preferencesConfig.authentication.client_id"</code>
+                </p>
+            </td>
+        </tr>
+        <tr>
+            <td>
+                <code>GPII_OAUTH2_AUTH_CLIENT_SECRET</code>
+            </td>
+            <td>
+                The client secret registered with the OAuth2 server.
+                Used to securely identify the client with the OAuth2.
+                Should be kept confidentially.
+                <p>
+                    Example:
+                    "client_secret_first_discovery"
+                </p>
+                <p>
+                    Option Path:
+                    <code>"preferencesConfig.authentication.client_secret"</code>
+                </p>
+            </td>
+        </tr>
+    </tbody>
+</table>
+
+### oauth2.json ###
+
+Builds on [environment.json](https://github.com/GPII/first-discovery-server/tree/master/src/config/environment.json), setting concrete values to configuration of the OAuth2 server that are known to exists for the GPII security layer.
```

Comment:
Ungrammatical and cumbersome sentence "that are known to exists for the"... it's not quite clear what is being said here. This whole sentence is quite unclear, including "Builds on". Should start by being clear what the relationship actually is - does one file contain defaults for the other? Or does the system support everything which is mentioned in one file plus the other? What is "the GPII security layer"?